doc/image_proc.rst

```diff
@@ -0,0 +1,351 @@
+.. include:: include/links.rst
+
+.. _image_proc:
+
+======================
+Basic Image Processing
+======================
+
+Here we describe the basic image processing steps performed by ``PypeIt``,
+specifically their order, salient details, and how to toggle specific steps
+using the :ref:`pypeit_file`.  This description is meant to be general to *all*
+spectrographs.  For instrument-specific advice, see :ref:`spec_details`.
+
+Unless otherwise stated, the relevant parameters governing basic image
+processing are kept by :class:`~pypeit.par.pypeitpar.ProcessImagesPar`; see
+:ref:`pypeitpar`.  Importantly, however, changing the default parameters for
+some frame types will lead to faults; see :ref:`proc_algorithm` and
+:ref:`workflow`.
+
+.. _overview:
+
+Overview
+========
+
+We provide a very general approach to image processing with hooks to toggle each
+step, as appropriate for different frame types and/or different instruments.
+Generally, we treat pixel values, :math:`p`, in an observed frame as containing
+the following components:
+
+.. math::
+
+    p = O + B + (C + D\ t_{\rm exp}) / \gamma
+
+where:
+
+    - :math:`O` is a frame-dependent bias estimate (in ADU) using image overscan
+      regions,
+    - :math:`B` is a longer-term pixel-by-pixel bias estimate (in ADU) using
+      bias images,
+    - the quantity :math:`C=c/s` is the number of electron counts excited by
+      photons hitting the detector,
+    - :math:`1/s` is an efficiency factor (one of many) that accounts for relative
+      throughput factors (see below) that can be measured from flat-field frames,
+    - :math:`D` is the rate at which the detector generates thermal electrons (in e-/s),
+    - :math:`t_{\rm exp}` is the exposure time in seconds, and
+    - :math:`\gamma` is the amplifier gain in e- per ADU.
```

Comment:
Can you consider making it so that is equation is in electrons and not in ADU? In other words, divide O and B by gamma, instead of C + Dt_exp?

Reply:
I prefer it this way, but let me know more about your rationale.  The reason I prefer it this way is because the combination of this equation and the next equation show how PypeIt converts the raw image values (p) to the reduced values (c) used throughout the rest of the code.  I.e., the equation for the counts in electrons (C) is the term in the brackets of the next equation.

Reply:
After re-reading it I agree with your approach. I modified the docs a bit so look over what I added and see if you are fine with it.